Feature/integration project frontend 05

pdrflp0 opened the pull request.

pdrflp0 committed:
### Added
   - CI File ".gitlab-ci.yml"

pdrflp0 committed:
### Added
   - Required dependencies for unit tests and tests with javaFX.
   - Unit testing of screencontroller methods.
   - Unit test of lines, models e categories of services.

### Changed
   - ScreenController adapted to pass unit tests.
   - DTO categories, DTO lines and DTO models adapted for unit tests.
   - Services from categories, lines and models adapted for unit testing.

### Removed
   -

gustavo-lopes-dev commented on src/main/java/br/eletra/frontend/controller/ScreenController.java:
```diff
@@ -18,19 +18,19 @@
 
     @FXML
-    private ComboBox<LineDTO> comboBoxLines;
+    ComboBox<LineDTO> comboBoxLines;
 
     @FXML
-    private TitledPane titledPaneModels;
+    TitledPane titledPaneModels;
 
     @FXML
-    private TreeView<LineDTO> treeView;
+    TreeView<Object> treeView;
 
     @FXML
-    private Accordion accordion;
+    Accordion accordion;
```
Favor, não deixar as varíaveis sem os modificadores. Como é usado para os testes, deixar todos como "protected", com esse modificador, é possível acessar as variaveis dentro das classes que compartilham o mesmo pacote

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/main/java/br/eletra/frontend/controller/ScreenController.java:
```diff
@@ -43,28 +43,29 @@ public void initialize(URL location, ResourceBundle resources) {
+    void openTreeView(LineDTO selectedLine) {
         titledPaneLines.setExpanded(false);
         titledPaneModels.setDisable(false);
         titledPaneModels.setExpanded(true);
 
-        List<CategoryDTO> categoryList = CategoryServices.getAllCategories(selectedLine);
-        TreeItem showTreeView = new TreeItem<>(selectedLine);
-        showTreeView.setExpanded(true);
+        List<CategoryDTO> categoryList = categoryServices.getAllCategories(selectedLine);
+        TreeItem<Object> rootItem = new TreeItem<>(new CategoryDTO(selectedLine.getLineName(), selectedLine.getId()));
+        rootItem.setExpanded(true);
 
-        categoryList.forEach((category) -> {
-            TreeItem<CategoryDTO> categoryItem = new TreeItem<>(category);
-            showTreeView.getChildren().add(categoryItem);
+        categoryList.forEach(category -> {
+            TreeItem<Object> categoryItem = new TreeItem<>(category);
+            rootItem.getChildren().add(categoryItem);
 
-            List<ModelDTO> modelList = ModelServices.getAllModels(category);
-            modelList.forEach((model) -> categoryItem.getChildren().add(new TreeItem(model)));
+            List<ModelDTO> modelList = modelServices.getAllModels(category);
+            modelList.forEach(model -> categoryItem.getChildren().add(new TreeItem<>(model)));
         });
-        treeView.setRoot(showTreeView);
+
+        treeView.setRoot(rootItem);
     }
```
Favor, não deixar os métodos sem os modificadores. Como é usado para os testes, deixar todos como "protected", com esse modificador, é possível acessar esses métodos dentro das classes que compartilham o mesmo pacote

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/controller/ScreenControllerTest.java:
Por favor, não esquecer de colocar a anotação After nas classes de testes. Ele é fundamental pra limpar as instâncias criadas na anotação Before e evitar problemas com memória

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/services/CategoryServicesTest.java:
Por favor, não esquecer de colocar a anotação After nas classes de testes. Ele é fundamental pra limpar as instâncias criadas na anotação Before e evitar problemas com memória

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/services/LineServicesTest.java:
Por favor, não esquecer de colocar a anotação After nas classes de testes. Ele é fundamental pra limpar as instâncias criadas na anotação Before e evitar problemas com memória

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/services/ModelServicesTest.java:
Por favor, não esquecer de colocar a anotação After nas classes de testes. Ele é fundamental pra limpar as instâncias criadas na anotação Before e evitar problemas com memória

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/controller/ScreenControllerTest.java:
```diff
@@ -0,0 +1,206 @@
+        controller.titledPaneModels.setContent(controller.treeView);
+    }
+
+    @Test
+    public void testInitializeController() {
+        // Given
+        Mockito.doNothing().when(controller).comboBoxSelect();
+        controller.titledPaneLines.setDisable(true);
+
+        // When
+        controller.initialize(null , null);
+
+        // Then
+        assertEquals(controller.titledPaneLines , controller.accordion.getExpandedPane());
+        assertTrue(controller.titledPaneModels.isDisable());
+        verify(controller).comboBoxSelect();
+    }
```
Na linha de código `controller.titledPaneLines.setDisable(true);` é uma falha no teste pois dentro do método initialize() ocorre o mesmo set, favor, corrigir essa falha

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/controller/ScreenControllerTest.java:
```diff
@@ -0,0 +1,206 @@
+        verify(controller, times(2)).openTreeView(controller.comboBoxLines.getItems().get(0));
+    }
+
+    @Test
+    public void testComboBoxSelect03() {
+        // Given
+        List<LineDTO> mockList = new ArrayList<>();
+        mockList.add(new LineDTO((short) 1, "Ares"));
+        mockList.add(new LineDTO((short) 2, "Cronos"));
+        Mockito.when(controller.lineServices.getAllLines()).thenReturn(mockList);
+        controller.comboBoxSelect();
+        controller.comboBoxLines.setValue(controller.comboBoxLines.getItems().get(1));
+
+        // When
+        controller.openTreeView(mockList.get(0));
+
+        // Then
+        verify(controller).openTreeView(controller.comboBoxLines.getItems().get(1));
+    }
+
+    @Test
+    public void testComboBoxSelect04() {
+        // Given
+        List<LineDTO> mockList = new ArrayList<>();
+        mockList.add(new LineDTO((short) 1, "Ares"));
+        Mockito.when(controller.lineServices.getAllLines()).thenReturn(mockList);
+        controller.comboBoxSelect();
+
+        // When
+        controller.openTreeView(mockList.get(0));
+
+        // Then
+        assertFalse(controller.comboBoxLines.getItems().isEmpty());
+    }
```
Não consegui compreender a necessidades desses testes e podem ser apagados

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev commented on src/test/java/br/eletra/frontend/services/CategoryServicesTest.java:
```diff
@@ -0,0 +1,71 @@
+        Gson mockGson = new Gson();
+        when(response.readEntity(String.class)).thenReturn(jsonResponse);
+        Type mockCategoryListType = new TypeToken<List<CategoryDTO>>() {}.getType();
+        List<CategoryDTO> mockCategoryList = mockGson.fromJson(response.readEntity(String.class), mockCategoryListType);
```
essa variável não é usada, favor apagar

pdrflp0 replied:
Corrigido ✅

gustavo-lopes-dev reviewed (changes requested):
Foi realizado o code review e foi identificado pontos para melhoria e/ou correção

pdrflp0 committed:
### Added
   - The "protected" modifiers in ScreenController variables.
   - "@After" annotation in "ScreenControllerTest" test classes.
   - "@After" annotation in "CategoryServicesTest" test classes.
   - "@After" annotation in "ModelServicesTest" test classes.
   - "@After" annotation in "LineServicesTest" test classes.

### Changed
   -
   -

### Removed
   - The line of code: "controller.titledPaneLines.setDisable(true);" in "ScreenControllerTest" as it was a failure.
   - 2 unit tests related to the "comboBox" method in "ScreenControllerTest" were not necessary.
   - Variable that was not used in "CategoryServicesTest", called "mockCategoryList".

pdrflp0 committed:
Merge branch 'develop' into feature/integration_project_frontend_05
